theano/tensor/tests/test_subtensor.py

```diff
@@ -337,6 +337,23 @@ def test_slice_symbol(self):
         ret = f()
         assert ret.shape == (1, 1, 4)
 
+    def test_ellipsis(self):
+        numpy_n = numpy.arange(24, dtype=self.dtype).reshape((2, 3, 4))
+        n = self.shared(numpy_n)
+        for slice_ in [numpy.index_exp[...],
+                       numpy.index_exp[..., 1],
+                       numpy.index_exp[1, ...],
+                       numpy.index_exp[..., 1, 2, 3],
+                       numpy.index_exp[1, ..., 2, 3],
+                       numpy.index_exp[1, 2, 3, ...],
+                       ]:
+            numpy_tval = numpy_n[slice_]
+            t = n[slice_]
+            self.assertTrue(isinstance(t.owner.op, Subtensor))
+            tval = self.eval_output_and_check(t)
```

Comment:
The error is that the optimized graph have have more then 1 node. Print it (you can modify the asser for that) It will help you understand what is going on. I have no idea just by code review.